docs/aws/audit/dynamodbmonitoring/rules/ddb_customer_kms_key_remediation.mdx

```diff
@@ -1,6 +1,228 @@
+   - In the "Manage encryption" settings, select "AWS Key Management Service (KMS)".
+   - Choose the Customer Master Key (CMK) you want to use from the dropdown list or create a new CMK if necessary.
+   - Save the changes to apply KMS encryption to your DynamoDB table.
+
+By following these steps, you can ensure that your DynamoDB tables are encrypted using KMS CMKs, enhancing the security of your data.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that your AWS DynamoDB tables use KMS Customer Master Keys (CMKs) for encryption, you can follow these steps using the AWS CLI:
+
+1. **Create a KMS Key**:
+   First, create a KMS key that will be used to encrypt your DynamoDB table.
+   ```sh
+   aws kms create-key --description "DynamoDB encryption key"
+   ```
+
+2. **Get the KMS Key ID**:
+   Retrieve the Key ID of the newly created KMS key.
+   ```sh
+   aws kms list-keys
+   ```
+
+3. **Enable Encryption on a New DynamoDB Table**:
+   When creating a new DynamoDB table, specify the KMS key for encryption.
+   ```sh
+   aws dynamodb create-table \
+       --table-name MyTable \
+       --attribute-definitions AttributeName=Id,AttributeType=S \
+       --key-schema AttributeName=Id,KeyType=HASH \
+       --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5 \
+       --sse-specification Enabled=true,SSEType=KMS,KMSMasterKeyId=<Your-KMS-Key-ID>
+   ```
+
+4. **Update an Existing DynamoDB Table to Use KMS Encryption**:
+   If you have an existing table, you can update it to use the KMS key for encryption.
+   ```sh
+   aws dynamodb update-table \
+       --table-name MyExistingTable \
+       --sse-specification Enabled=true,SSEType=KMS,KMSMasterKeyId=<Your-KMS-Key-ID>
+   ```
+
+Replace `<Your-KMS-Key-ID>` with the actual Key ID obtained from the `list-keys` command. These steps will ensure that your DynamoDB tables are encrypted using KMS CMKs.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that AWS DynamoDB tables use KMS Customer Master Keys (CMKs) for encryption, you can use the AWS SDK for Python (Boto3). Below are the steps to prevent this misconfiguration:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+3. **Create a Python Script**:
+   Write a Python script to create a DynamoDB table with KMS CMK encryption enabled.
+
+4. **Run the Script**:
+   Execute the script to create the DynamoDB table with the specified encryption settings.
+
+Here is a sample Python script to create a DynamoDB table with KMS CMK encryption:
+
+```python
+import boto3
+
+# Initialize a session using Amazon DynamoDB
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>